### PR TITLE
Ln cted 2381

### PR DESCRIPTION
**Pull request by loyisongq**

### CODE REVIEW

File: classes/controller/edit_controller.php [hunk 1]
```diff
@@ -175,6 +175,8 @@ public function add_discussion_action() {
                 'timestart'     => $timestart,
                 'timeend'       => $timeend,
             );
+
+
```
Comment (BrendonPretorius) [inline classes/controller/edit_controller.php:179]: No need for the double lines

File: classes/controller/edit_controller.php [hunk 2]
```diff
@@ -161,7 +161,7 @@ public function add_discussion_action() {
             $context = $PAGE->context;
             $course  = $PAGE->course;
 
-            if ($groupid == 0) {
+            if ($groupid === 0) {
```
Comment (BrendonPretorius) [inline classes/controller/edit_controller.php:164]: Are you sure that you want the strict comparison? Will it always be same type? If not an int, what happens?

File: classes/service/discussion_service.php [hunk 3]
```diff
@@ -70,6 +70,7 @@ public function __construct(\moodle_database $db = null) {
     public function handle_add_discussion($course, $cm, $forum, $context, array $options, $posttomygroups = false) {
         global $PAGE, $OUTPUT;
 
+
```
Comment (BrendonPretorius) [inline classes/service/discussion_service.php:73]: No need for this newline

File: classes/service/discussion_service.php [hunk 4]
```diff
@@ -81,18 +82,22 @@ public function handle_add_discussion($course, $cm, $forum, $context, array $opt
...
                 }
             }
-        } elseif (isset($_POST['groupinfo'])) {
-            foreach ($options['groupid'] as $groupid) {
-                $groupstopostto[] = $groupid;
-            }
         } else {
-            $groupstopostto[] = $options['groupid'];
+            $groupids = $options['groupid'];
+            //if ias array
```
Comment (BrendonPretorius) [inline classes/service/discussion_service.php:87]: Typo in comment and not according to moodle comment standards

**Review by BrendonPretorius — CHANGES_REQUESTED** [review]

File: classes/controller/edit_controller.php (hunk 2 above)
Comment (loyisongq) [reply classes/controller/edit_controller.php:164]: yep it will always return an int from the groups table, but updating to equals to comparison

**Comment by loyisongq** [comment]
Screen Shot 2022-04-13 at 1 25 16 PM
Screen Shot 2022-04-13 at 1 21 23 PM
Screen Shot 2022-04-13 at 1 09 28 PM
![Screen Shot 2022-04-13 at 1 28 29 PM
](https://user-images.githubusercontent.com/71312749/163170357-7dccc62e-d230-44d1-9c25-f869cf661441.png)

renderer.php and classes/post_form.php no errors detected on code i added

**Review by BrendonPretorius — APPROVED** [review]

**Pull request merged**